### DISCUSSION TITLE
How should I benefit JIT with Ajax

### DISCUSSION THREAD

**Discussion by hoppergee** [tailwindlabs/tailwindcss#4455] [Help] [answered]
Say the browser get a placeholder page like this:

```html
<head>
...
</head>
<body>
  <div class="m-1">Hello</div>
  <div id="placeholder"></div>
  <script>
    # Fetch some content with Ajax and replace the placeholder
  </script>
</body>
```
(The JIT will create "m-1" class for us on the fly)

The response body of the Ajax request is something like below:

```html
<div class="p-1">World</div>
```

Then our HTML body becomes:

```html
<body>
  <div class="m-1">Hello</div>
  <div class="p-1">World</div>
  <script>
    ...
  </script>
</body>
```
The "p-1" definition is missing, because we only get a part of raw HTML code, JIT doesn't has an opportunity to generate it.

The JIT seems does not offer a way to handle this kind of scenario. What should I do to benefit JIT with Ajax?

**Comment by therealgilles** [comment, marked as answer]
@hoppergee: This is not specific to Ajax. Tailwindcss is a pre-processor, which means it runs beforehand and on static code. If you manipulate the DOM at runtime, there is no way it can know. You have a few options:
1) Add the classes you need in the safelist file.
2) Put the html you are planning to add in a template Tailwindcss can parse.
3) Add the classes to your stylesheet in the non-purge zone.

**Reply by hoppergee** [reply]
Thank you @therealgilles. I think the option 1 is what I want.
